cinex/PageModels/MainPageModel.cs

```diff
@@ -0,0 +1,213 @@
+﻿using cinex.Models;
+using cinex.Services;
+using FreshMvvm;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+
+namespace cinex.PageModels
+{
+    [PropertyChanged.AddINotifyPropertyChangedInterface]
+    public class MainPageModel : FreshBasePageModel
+    {
+        IAppService _appService;
+
+        private int NextPage { get; set; } = 1;
+
+        private int TotalPages { get; set; } = 1;
+
+        public bool IsLoading { get; set; }
+
+        public string Title { get; set; }
+
+        public ObservableCollection<UpcomingModel> Upcomings { get; }
+
+        public List<UpcomingModel> OriginalUpcomings { get; set; }
+
+        public List<GenreModel> Genres { get; }
+
+        public bool ShowLoadMore { get; set; }
+
+        private string _searchString;
+        public string SearchString
+        {
+            get => _searchString;
+            set
+            {
+                if (!EqualityComparer<string>.Default.Equals(_searchString, value))
+                {
+                    _searchString = value;
+                    FilterUpcomings();
```

Comment:
Avoid doing too much work inside `property setters`

Reply:
Right, I removed the unnecessary part, but I keep the call to the FilterUpcomingsCommand.